src/constants.rs

```diff
@@ -18,33 +18,49 @@ impl Paths {
     pub fn initialize() -> std::io::Result<Paths> {
         let pd = ProjectDirs::from("com.github.target.lorri", "lorri", "lorri")
             .expect("Could not determine lorri project/cache directories, please set $HOME");
-        let create_dir = |dir: PathBuf| -> std::io::Result<PathBuf> {
+        let create_dir = |dir: AbsPathBuf| -> std::io::Result<AbsPathBuf> {
             std::fs::create_dir_all(&dir).and(Ok(dir))
         };
+
+        // TODO: return as good error value
+        assert!(
+            pd.cache_dir().is_absolute(),
+            "Your cache directory is not an absolute path! It is: {}",
```

Comment:
Is there anything the user could do about this error message when they see it? `pd` comes from `ProjectDirs::from`, so it seems to me like that method should ensure this is a "good" path. Perhaps `ProjectDirs::cache_dir` could return an `AbsPathBuf`?

Reply:
`ProjectDirs::cache_dir` comes from a different library. I opted for a `panic!()` for now, because if the paths return by `ProjectDirs` are not absolute, we have a problem (because we want to ensure that nix files are absolute).